### PR TITLE
Migrate Google Tag Manager to a plugin

### PR DESCRIPTION
**Pull request by gris**
## What's the purpose of this pull request?
Removes google tag manager from `gatsby-theme-store` and make it available as a plugin.

## How it works? 
<em>Tell us the role of the new feature, or component, in its context.</em>

## How to test it?
<em>Describe the steps with bullet points. Is there any external link that can be used to better test it or an example?</em> 

## References
<em>Spread the knowledge: is there any content you used to create this PR that is worth sharing?</em>  

<em>Extra tip: adding references to related issues or mentioning people important to this PR may be good for the documentation and reviewing process</em>

### CODE REVIEW

File: packages/gatsby-plugin-google-tag-manager/src/pixel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import React, { Fragment, useMemo } from 'react'
+import type { FC } from 'react'
+import { isServer } from '@vtex/gatsby-theme-store/src/utils/env'
+import { useLazyScript } from '@vtex/gatsby-theme-store/src/sdk/lazyScript/useLazyScript'
+import { usePixelEvent } from '@vtex/gatsby-theme-store/src/sdk/pixel/usePixelEvent'
+import { once } from '@vtex/gatsby-theme-store/src/sdk/once'
+
+import { handler } from './handler'
+
+const DEFAULT_DATALAYER_CONFIG = [{ 'gtm.blacklist': ['html'] }]
```
Comment (klzns) [inline packages/gatsby-plugin-google-tag-manager/src/pixel/index.tsx:10]: Maybe have an option to disable that too

File: packages/gatsby-plugin-google-tag-manager/src/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+// noop
```
Comment (klzns) [inline packages/gatsby-plugin-google-tag-manager/src/index.ts:1]: Why is that needed?

**Review by klzns — APPROVED** [review]

File: packages/gatsby-plugin-google-tag-manager/src/index.ts (hunk 2 above)
Comment (gris) [reply packages/gatsby-plugin-google-tag-manager/src/index.ts:1]: I thought this would be needed if a user `require` my plugin. Not 100% sure about this https://www.gatsbyjs.com/docs/files-gatsby-looks-for-in-a-plugin/

Comment (gris) [reply packages/gatsby-plugin-google-tag-manager/src/index.ts:1]: image

File: packages/gatsby-plugin-google-tag-manager/src/pixel/index.tsx (hunk 1 above)
Comment (gris) [reply packages/gatsby-plugin-google-tag-manager/src/pixel/index.tsx:10]: Nice! I will do that in a future contribution. #vemai

File: packages/gatsby-plugin-google-tag-manager/src/gatsby-browser.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+const GTMProvider = require('./src/pixel/index').default
+
+export const wrapRootElement = (
+  { element }: WrapRootElementBrowserArgs,
+  { gtmId }: Props
+) => (
+  <GTMProvider
+    gtmId={gtmId}
+    // Overwrites the default blacklist config
+    dataLayerConfig={[]}
```
Comment (tlgimenes) [inline packages/gatsby-plugin-google-tag-manager/src/gatsby-browser.tsx:13]: Please make it possible to pass this prop via plugin config and make the default our DEFAULT_DATALAYER_CONFIG

File: packages/gatsby-plugin-google-tag-manager/package.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  "devDependencies": {
+    "@vtex/gatsby-theme-store": "^0.232.0",
+    "cross-env": "^7.0.2",
+    "gatsby": "^2.24.37",
+    "typescript": "^4.1.2"
+  },
+  "peerDependencies": {
+    "@vtex/gatsby-theme-store": "^0.232.0",
+    "gatsby": "^2.24.37",
+    "graphql": "^14.0.0 || ^15.0.0"
```
Comment (tlgimenes) [inline packages/gatsby-plugin-google-tag-manager/package.json:29]: do we really need to depend on graphql ?

File: packages/gatsby-plugin-google-tag-manager/package.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "@vtex/gatsby-plugin-google-tag-manager",
```
Comment (tlgimenes) [inline packages/gatsby-plugin-google-tag-manager/package.json:2]: isn't it better to call it of gatsby-plugin-pixel-tag-manager ?

File: packages/gatsby-plugin-google-tag-manager/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "outDir": ".",
+    "alwaysStrict": false,
+    "jsx": "react"
+  },
+  "include": ["./src/**/*.ts", "./src/**/*.tsx", "src/gatsby-node.js"],
```
Comment (tlgimenes) [inline packages/gatsby-plugin-google-tag-manager/tsconfig.json:8]: can you exclude this "src/gatsby-node.js" ?

**Review by tlgimenes — CHANGES_REQUESTED** [review]
Minor changes. Also, can you add a schema for this plugin using the gatsby-node.js api ? Also, please add a [readme](https://www.gatsbyjs.com/contributing/how-to-write-a-plugin-readme/)

File: packages/gatsby-plugin-google-tag-manager/src/gatsby-browser.tsx (hunk 3 above)
Comment (tlgimenes) [reply packages/gatsby-plugin-google-tag-manager/src/gatsby-browser.tsx:13]: Actually, you could do a spread of plugin props

File: packages/gatsby-plugin-google-tag-manager/package.json (hunk 5 above)
Comment (gris) [reply packages/gatsby-plugin-google-tag-manager/package.json:2]: Humm, I've didn't find any references to the google-tag-manager being called a script.

With facebook they seem to use that name explicitly https://developers.facebook.com/docs/facebook-pixel/implementation

I was thinking about leveraging this opportunity to rethink how we name things. Since we don't have pixel builder/server anymore.

What are your thoughts @tlgimenes @klzns ?

Comment (klzns) [reply packages/gatsby-plugin-google-tag-manager/package.json:2]: I agree, pixel was a term that we used but is not really that common. I would keep the name as is

File: packages/gatsby-plugin-google-tag-manager/package.json (hunk 4 above)
Comment (gris) [reply packages/gatsby-plugin-google-tag-manager/package.json:29]: Removing it.

File: packages/gatsby-plugin-google-tag-manager/src/gatsby-browser.tsx (hunk 3 above)
Comment (gris) [reply packages/gatsby-plugin-google-tag-manager/src/gatsby-browser.tsx]: Done

File: packages/gatsby-plugin-google-tag-manager/README.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+This plugin integrates `gatsby-theme-store` with Google Tag Manager.
+
+## How to install
+
+Just run `npm i @vtex/gatsby-plugin-google-tag-manager` in the root of your code.
+
+## Available options (if any)
+
+- `gtmId` **required**: this options informs to Google Tag Manager what is the ID of your account
```
Comment (klzns) [inline packages/gatsby-plugin-google-tag-manager/README.md:11]:
```suggestion
- `gtmId` **required**: the ID of the Google Tag Manager container.
```

File: packages/gatsby-plugin-google-tag-manager/README.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+- `dataLayerConfig` **optional**: this options informs what are the options that you want to have in your data layer. By default we pass `[{ 'gtm.blacklist': ['html'] }]`. If you want to be able to use custom HTML tags with Google Tag Manager, use `[]` in this option.
```
Comment (klzns) [inline packages/gatsby-plugin-google-tag-manager/README.md:12]:
```suggestion
- `dataLayerConfig` **optional**: this options informs what are the options that you want to have in your data layer. By default we pass `[{ 'gtm.blacklist': ['html'] }]`. If you want to be able to use Custom HTML tags in Google Tag Manager, use `[]` in this option.
```

File: packages/gatsby-plugin-google-tag-manager/README.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+## When do I use this plugin?
+
+When you want to use Google Tag Manager.
```
Comment (klzns) [inline packages/gatsby-plugin-google-tag-manager/README.md:16]:
```suggestion
When you want to use Google Tag Manager and have the Google Enhanced Ecommerce events.
```

**Review by klzns — CHANGES_REQUESTED** [review]

**Review by klzns — APPROVED** [review]

**Review by tlgimenes — APPROVED** [review]
